### PR TITLE
Mouse button events

### PR DESCRIPTION
**Pull request by jobrittain**
## Changes

* Added events to capture press/release input  from the left, middle and right buttons on the mouse
* Fixed GLFW include in cmake
* Added 'magic enum' header only library to allow good enum stringification

## Screenshots

563c915c-5559-42bd-8d15-1c94f5d21927

### CODE REVIEW

File: .editorconfig [hunk 1]
```diff
@@ -0,0 +1,3 @@
+[*.{h,cpp}]
+indent_style = space
+indent_size = 2
```
Comment (hypzeh) [inline .editorconfig]: Very nice idea, definitely need this.

Needs an newline at the end of file for github though.

Comment (hypzeh) [reply .editorconfig]: Is there a rule which helps with the newline ending?

Comment (jobrittain) [reply .editorconfig]: Cool, I actually forgot about that. I will add a new line.

File: src/Onyx.Engine/System/Events/MouseClickEvent.cpp [hunk 2]
```diff
@@ -0,0 +1,25 @@
+#include "MouseClickEvent.h"
+
+Onyx::Engine::System::Events::MouseClickEvent::MouseClickEvent(MouseButtonType type, MouseButtonAction action)
```
Comment (hypzeh) [inline src/Onyx.Engine/System/Events/MouseClickEvent.cpp]: I've been wrapping implementations inside a namespace so I don't have to type them out all the time, is there any advantages to this?

File: .editorconfig (hunk 1 above)
Comment (jobrittain) [reply .editorconfig]: > Is there a rule which helps with the newline ending?

I have just found out there is.

File: src/Onyx.Engine/System/Events/MouseClickEvent.cpp (hunk 2 above)
Comment (jobrittain) [reply src/Onyx.Engine/System/Events/MouseClickEvent.cpp]: Oh right, I hadn't noticed. I ususally just generate them using intellisense from the header.

**Review by hypzeh — APPROVED** [review]
That demo looks great, I was thinking it would be much more work!

There are a few files that need a line ending for github.

**Comment by jobrittain** [comment]
> That demo looks great, I was thinking it would be much more work!

The most stressful thing about it was dealing with the glfw header not being found, and subsequently figuring out what adjustment to make to cmakelists.

**Pull request merged**